secretservice/dh_ietf1024_sha256_aes128_cbc_pkcs7.go

```diff
@@ -0,0 +1,145 @@
+// This file implements a basic Diffie-Hellman for groups with modular
+// exponentiation operators. In particular, it is used in this package
+// to implement the Diffie-Hellman KEX over the Second Oakley Group.
+// meant only for use for securing the channel to the D-Bus Secret Service.
+// Much of the code in this file is derived from package
+// golang.org/x/crypto/ssh:kex.go, and is replicated here because the relevant
+// variables and methods are not exported or easily accessible.
+// Note that this protocol is NOT authenticated, NOT secure against malleation
+// and is NOT CCA2-secure. It is only meant to hide the D-Bus messages from any
+// system services that may be logging everything.
+
+package secretservice
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	cryptorand "crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"math/big"
+
+	errors "github.com/pkg/errors"
+	"golang.org/x/crypto/hkdf"
+)
+
+type dhGroup struct {
+	g, p, pMinus1 *big.Int
+}
+
+var bigOne = big.NewInt(1)
+
+func (group *dhGroup) NewKeypair() (private *big.Int, public *big.Int, err error) {
+	for {
+		if private, err = rand.Int(cryptorand.Reader, group.pMinus1); err != nil {
+			return nil, nil, err
+		}
+		if private.Sign() > 0 {
+			break
+		}
+	}
+	public = new(big.Int).Exp(group.g, private, group.p)
+	return private, public, nil
+}
+
+func (group *dhGroup) diffieHellman(theirPublic, myPrivate *big.Int) (*big.Int, error) {
+	if theirPublic.Cmp(bigOne) <= 0 || theirPublic.Cmp(group.pMinus1) >= 0 {
+		return nil, errors.New("ssh: DH parameter out of bounds")
+	}
+	return new(big.Int).Exp(theirPublic, myPrivate, group.p), nil
+}
+
+func RFC2409SecondOakleyGroup() *dhGroup {
```

Comment:
golint is going to complain that RFC2409SecondOakleyGroup is exported but is not returning an exported type.